core/server/master/src/test/java/alluxio/master/file/RpcContextTest.java

```diff
@@ -105,6 +108,46 @@ public void journalContextThrows() throws Throwable {
     checkClose(jcException);
   }
 
+  @Test
+  public void testCallTrackers() throws Throwable {
+    InternalOperationContext opCtx = new InternalOperationContext();
+    // Add a call tracker that's always cancelled.
+    opCtx = opCtx.withTracker(new CallTracker() {
+      @Override
+      public boolean isCancelled() {
+        return true;
+      }
+
+      @Override
+      public Type getType() {
+        return Type.GRPC_CLIENT_TRACKER;
+      }
+    });
+    // Add a call tracker that's never cancelled.
+    opCtx = opCtx.withTracker(new CallTracker() {
+      @Override
+      public boolean isCancelled() {
+        return false;
+      }
+
+      @Override
+      public Type getType() {
+        return Type.STATE_LOCK_TRACKER;
+      }
+    });
+    // Create RPC context.
+    RpcContext rpcCtx = new RpcContext(mMockBDC, mMockJC, opCtx);
+    // Verify the RPC is cancelled due to tracker that's always cancelled.
+    assertTrue(rpcCtx.isCancelled());
+    try {
+      // Verify cancellation throws.
+      rpcCtx.throwIfCancelled();
+      fail("Call should have been cancelled.");
+    } catch (RuntimeException e) {
```

Comment:
This is a good case for the `ExpectedException` rule since there aren't any assertions afterwards ;)